pkg/ingester/ingester.go

```diff
@@ -1297,6 +1308,9 @@ func (i *Ingester) Push(ctx context.Context, req *cortexpb.WriteRequest) (*corte
 
 		// Look up a reference for this series.
 		tsLabels := cortexpb.FromLabelAdaptersToLabels(ts.Labels)
+		if i.isLabelSetOutOfOrder(tsLabels) {
+			return nil, wrapWithUser(errors.Errorf("out-of-order label set found when push: %s", tsLabels), userID)
```

Comment:
Should we also add a metric for total number of out of order requests?

Reply:
Added